simplifier/operations/bitwise.py

```diff
@@ -175,6 +175,39 @@ def negate(self, level: Optional[int] = None, predecessors: Optional[List[Operat
         self.world.replace(self, new_op)
         return new_op
 
+    def get_equivalent_condition_of_negated_type(self, operation: Union[BitwiseAnd, BitwiseOr]) -> Union[BitwiseAnd, BitwiseOr]:
+        """
+        Create the new condition that is equivalent to self and replace self by it.
+
+        - if self is Or-operation and operation is a conjunction
+            self = (a1 & a2 & ... & ak) | l1 | l2 | ... | lh, where operation = (a1 & a2 & ... & ak),
+            remaining_conditions = [l1, l2, ..., lh] and each li is a condition
+            -> new_condition = (a1 | l1 | ... |lh) & (a2 | l1 | ... |lh) & ... & (ak | l1 | ... |lh)
+        -if self is And-operation and operation is a disjunction
+            self = (a1 | a2 | ... | ak) & l1 & l2 & ... & lh, where disjunction = (a1 | a2 | ... | ak),
+            remaining_conditions = [l1, l2, ..., lh] and each li is a condition
+            -> new_condition = (a1 & l1 & ... & lh) | (a2 & l1 & ... & lh) | ... | (ak & l1 & ... & lh)
+        """
+        conditions = []
+        for literal in operation.operands:
+            new_operands = [literal, *(cond.copy_tree() for cond in self.operands if hash(cond) != hash(operation))]
```

Comment:
Could you elaborate on the `if hash(cond) != hash(operation))` part?
It seems to check if a operation contains itself as an operand

Reply:
Okay, perhaps the names are not optimal. When you have a look at the docstring, the `A` is the input-operation `operation`. The operation `A` is an operand of `self`, the operation we want to modify. Now, the new operands consist of one literal of `A`, the input operation, and all other operands of self, except `A`. Therefore, we check this using the hash.

Reply:
So `self` may contain `operation`? Guess that kinda makes sense